server/src/routes/signin.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * A router that handles user signin process
+ * It responds to POST requests on endpoint of "/api/auth/signin"
+ * This router takes in user "email" and "password" as its arguments
+ * and generates a JWT for the user object that contains "id" and "email" field inside the payload
+ * The jwt is then sent to user in form of cookies
+ */
+
+const router = require("express").Router();
+const { body } = require("express-validator");
+const db = require("../../database/models/index");
+const Password = require("../utils/password");
+const BadRequestError = require("../errors/bad-request-error");
+const validateRequest = require("../middleware/validate-request");
+const jwt = require("jsonwebtoken");
+require("dotenv").config();
+
+router.post(
+  "/api/auth/signin",
+  [
+    body("email").isEmail().withMessage("Email must be valid!"),//Validation step for Email
+    body("password").trim().notEmpty().withMessage("Password must be valid!"),// Validation Step for password
+  ],
+  validateRequest,//Middleware to process any errors with the validation stuff
+  async (req, res) => {
+    const { email, password } = req.body;
+
+    // Check if a user exists with a given email
+    const existingUser = await db.User.findOne({
+      where: {
+        email,
+      },
+    });
+
+    // If the user is not found then it throws a new BadRequestError which is handled by error-handler middleware
+    // The BadRequest error has a status code of 400 and returns the message inside the constructor
+    if (!existingUser) {
+      throw new BadRequestError("Invalid Email or Password!");
+    }
+
+    // Boolean variable that sees if the password in the DB matches the one provided by user
+    const passwordsMatch = await Password.compare(
+      existingUser.password,
+      password
+    );
+
+    // If the passwords don't match throw another BadRequestError
+    if (!passwordsMatch) {
+      throw new BadRequestError("Invalid Email or Password!");
+    }
+
+    // Finally if the user exists and passwords match here is where payload of jwt is assigned
+    // To sign the JWT key we need to have and environment variable of JWT_KEY
+    const userJWT = jwt.sign(
+      {
+        id: existingUser.id,
+        email: existingUser.email,
+      },
+      process.env.JWT_KEY
+    );
+
+    // Set a cookie called jwt inside the session
+    req.session = {
+      jwt: userJWT,
+    };
```

Comment:
Did you confirm if you use express sessions it can handle multiple users to sign up at the same time?

Reply:
I forgot my project was based off of cookie-session and not express-session. I tried using two instances of a rest client and it works great on both.